lib/perl/Genome/Model/Tools/Analysis/Coverage/MergeReadcounts.pm

```diff
@@ -237,7 +239,17 @@ sub execute {
 	else
 	{
 	    my @new = split /\t/, $line;
-	    die "Inconsistent annotation format found: " . $variant unless scalar(@header) == scalar(@new);
+	    
+	    die "Inconsistent table format found in the header line: " . $variant unless scalar(@header) == scalar(@new);
+	    
+	    # double-checks whether they share the same column header
+	    for (my $i=0; $i<@header; $i ++)
```

Comment:
Same here.  Nothing wrong at all but here's an alternative using `List::MoreUtils`' [all][1] and [pairwise][2].
```
use List::MoreUtils qw(all pairwise);

unless (all { $_ } pairwise { $a eq $b } @header, @new) {
    die ...
}
```

Both [List::Util][3] and [List::MoreUtils][4] are very handy libraries.  Do be aware our version is a bit out of date so viewing the documentation for our installed version would be better, e.g. `perldoc List::MoreUtils`.

[1]: https://metacpan.org/pod/List::MoreUtils#all-BLOCK-LIST
[2]: https://metacpan.org/pod/List::MoreUtils#pairwise-BLOCK-ARRAY1-ARRAY2
[3]: https://metacpan.org/pod/List::Util
[4]: https://metacpan.org/pod/List::MoreUtils